internal/subscriber/subscriberimpl_test.go

```diff
@@ -0,0 +1,329 @@
+package subscriber
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/golang/protobuf/proto"
+	mockBS "github.com/razorpay/metro/internal/brokerstore/mocks"
+	mocks "github.com/razorpay/metro/internal/node/mocks/repo"
+	"github.com/razorpay/metro/internal/offset"
+	"github.com/razorpay/metro/internal/subscription"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	mockMB "github.com/razorpay/metro/pkg/messagebroker/mocks"
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	subID      string = "subscriber-id"
+	subName    string = "subscription-name"
+	topic      string = "primary-topic"
+	retryTopic string = "retry-topic"
+	partition  int32  = 0
+)
+
+var ticker = time.NewTicker(3 * time.Second)
```

Comment:
A generic ticker will send out an event on the channel every 3 seconds. This is not relative to the test start time. Avoid using generic ticker and scope them based on the test execution.